src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -459,8 +453,19 @@ final case class AgreementApiServiceImpl(
     val nextStateByAttributes = AgreementStateByAttributesFSM.nextState(agreement, eService, consumer)
     val suspendedByPlatform   = suspendedByPlatformFlag(nextStateByAttributes)
     val newState              = agreementStateByFlags(nextStateByAttributes, None, None, suspendedByPlatform)
+
+    def calculateStamps(state: AgreementState, stamp: Stamp): Future[Stamps] = state match {
+      case AgreementState.PENDING   => Future.successful(agreement.stamps.copy(submission = stamp.some))
+      case AgreementState.ACTIVE    =>
+        Future.successful(agreement.stamps.copy(submission = stamp.some, activation = stamp.some))
+      case AgreementState.SUSPENDED =>
+        Future.successful(agreement.stamps.copy(submission = stamp.some, activation = stamp.some))
```

Comment:
from draft an agreement cannot go to suspended, which case should this cover?

Reply:
I fixed it. Thanks!